graphql_compiler/tests/test_compiler.py

```diff
@@ -7050,7 +7109,23 @@ def test_fold_and_optional(self):
                 )
             ])}
         '''
-        expected_sql = NotImplementedError
+        expected_sql = '''
+        SELECT
+            [Animal_1].name AS animal_name,
+            coalesce(folded_subquery_1.fold_output_1, ARRAY[]::VARCHAR[]) AS child_names_list,
+            [Animal_2].name AS parent_name
+        FROM db_1.schema_1.[Animal] AS [Animal_1]
+        LEFT OUTER JOIN (
+            SELECT
+                array_agg([Animal_1].name) AS fold_output_1,
+                [Animal_1].parent AS parent
+            FROM db_1.schema_1.[Animal] AS [Animal_1]
+            GROUP BY [Animal_1].parent
+        ) AS folded_subquery_1
+        ON [Animal_1].uuid = folded_subquery_1.parent
+        LEFT OUTER JOIN
+            db_1.schema_1.[Animal] AS [Animal_2]
+        ON [Animal_1].uuid = [Animal_2].parent'''
```

Comment:
```
graphql_input = '''{
    Animal {
        name @output(out_name: "animal_name")
        out_Animal_ParentOf @fold {
            name @output(out_name: "child_names_list")
        }
        in_Animal_ParentOf @optional {
            name @output(out_name: "parent_name")
        }
    }
}'''
```

Reply:
I think line 7128 should read `ON [Animal_1].parent = [Animal_2].uuid`

Reply:
This might be a merge conflict issue. I pushed a change to master correcting the direction of the `Animal_ParentOf` edge. I'm guessing the tests are incorrect (because of the merge conflict), not your code. You can see that the tests are failing.